code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,259 @@
+/obj/item/inflatable
+	name = "inflatable wall"
+	desc = "A folded membrane which rapidly expands into a large cubical shape on activation."
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "folded_wall"
+	w_class = 3
+
+/obj/item/inflatable/attack_self(mob/user)
+	playsound(loc, 'sound/items/zip.ogg', 75, 1)
+	to_chat(user, "<span class='notice'>You inflate [src].</span>")
+	var/obj/structure/inflatable/R = new /obj/structure/inflatable(user.loc)
+	src.transfer_fingerprints_to(R)
+	R.add_fingerprint(user)
+	qdel(src)
+
+/obj/structure/inflatable
+	name = "inflatable wall"
+	desc = "An inflated membrane. Do not puncture."
+	density = 1
+	anchored = 1
+	opacity = 0
+	CanAtmosPass = ATMOS_PASS_DENSITY
+
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "wall"
+
+	var/health = 50.0
+
+/obj/structure/inflatable/New(loc)
+	..()
+	density = 1
+	air_update_turf(1)
+
+/obj/structure/inflatable/Destroy()
+	density = 0
+	air_update_turf(1)
+	return ..()
+
+/obj/structure/inflatable/CanPass(atom/movable/mover, turf/target, height=0)
+	return 0
+
+/obj/structure/inflatable/bullet_act(var/obj/item/projectile/Proj)
+	health -= Proj.damage
+	..()
+	if(health <= 0)
+		deflate(1)
+	return
+
+/obj/structure/inflatable/ex_act(severity)
+	switch(severity)
+		if(1.0)
+			qdel(src)
+			return
+		if(2.0)
+			deflate(1)
+			return
+		if(3.0)
+			if(prob(50))
+				deflate(1)
+				return
+
+/obj/structure/inflatable/blob_act()
+	deflate(1)
+
+/obj/structure/inflatable/attack_hand(mob/user as mob)
+	add_fingerprint(user)
+	return
+
+/obj/structure/inflatable/attackby(obj/item/weapon/W as obj, mob/user as mob, params)
+	if(!istype(W))
+		return
+	if(W.is_sharp() || is_pointed(W))
+		visible_message("<span class='danger'>[user] pierces [src] with [W]!</span>")
+		deflate(1)
+	if(W.damtype == BRUTE || W.damtype == BURN)
+		hit(W.force)
+		..()
+	return
+
+/obj/structure/inflatable/proc/hit(var/damage, var/sound_effect = 1)
+	health = max(0, health - damage)
+	if(sound_effect)
+		playsound(loc, 'sound/effects/Glasshit.ogg', 75, 1)
+	if(health <= 0)
+		deflate(1)
+
+/obj/structure/inflatable/AltClick()
```

Comment:
I don't know if this is a good idea I just prefer the right-click verb, makes for less ambiguous accidents and people claiming misclicks.